### ISSUE TITLE
Scale slider - [new feature request]

### ISSUE THREAD

**Issue by davitgigilashvili** [khaivngo/QuickEval#154]
Is it possible to add a scale slider, primarily for magnitude estimation experiments? So, the experiment will look as follows: the stimulus is presented to an observer and the observer should rate this stimulus by moving a scale slider horizontally. The example how this might look can be found in the attached file. 
![slider_example](https://user-images.githubusercontent.com/89779831/131335770-0b602c77-ecee-49df-b390-a7445559a0f4.png)
The range of the slider scores (e.g. 0-100 or 0-10 etc.) should be customizable. This will yield a new type of experiment "Magnitude Estimation". 

The slider can be used for matching type of experiments as well. In this experiment, two stimuli will be displayed. One is fixed and the other one is variable. The observer moves the slider to change the properties of the variable image (so, what actually is happening is that new images are loaded in the variable image frame from a priori provided image set) and stops when considers that fixed and variable images match in appearance. This will yield a new type of experiment "Matching". 

It would be ideal, if both are implemented, but between those two, the first one, "Magnitude Estimation" is more important and urgent. 

Thanks, 
Davit

**Comment by disastersystem** [comment]
I can see both of these being useful features.

I think I'll be able to complete the "Magnitude Estimation" slider experiment this week.

A "Matching" experiment will take some more work. But I'll get to it eventually.

**Comment by davitgigilashvili** [comment]
That's excellent. Yes, magnitude estimation is more important for the time being. 

Thank you very much for your prompt reply, Robin!

**Comment by disastersystem** [comment]
The Magnitude Estimation experiment has been finished for a couple of days. Let me know if anything can be improved. I would recommend doing a quick test to see if you get correct result data, just to be sure.

I won't mark this issue as completed until the matching experiment is finished as well.

**Comment by davitgigilashvili** [comment]
Hi again. 

Thank you very much for completing this so promptly. I very much appreciate it! 

There are two issues I identified:

1. **Sampling of scale labels**

The scale looks perfect when there are few levels on it, e.g. 0 to 12. 

![image](https://user-images.githubusercontent.com/89779831/132832976-a29b2e47-e6a6-4fab-81f6-f9ac60d22b8d.png)

But when there are many levels on the scale, say 0-100, the labels on the scale get written over each other and looks kinda annoying. Maybe, when there are more than 10 or 20 levels, not all labels need to be displayed below the scale? There can be some principle for sampling. E.g. when we have a scale from 0 to 100, the scale can show every 10th label: 0-10-20-30...-100.

![image](https://user-images.githubusercontent.com/89779831/132833254-55c0828c-57c6-489a-9cc8-265596824384.png)


2. **Automatically generated statistics missing**
The exported data in the .CVS file are correct, but statistics do not get generated. It says "coming soon"

![image](https://user-images.githubusercontent.com/89779831/132831320-779acacf-0299-42fa-ac16-372009ca2328.png)

Indeed, it's not urgent and essential at this stage, as I can calculate the statistics in Excel myself, but would be nice at some point to have a few summary statistics directly at QuickEval — somewhat similar to what it is for category judgment experiment (mean, median, Z-scores etc.) 

Kind regards, 
Davit

**Comment by disastersystem** [comment]
Ah yes. The scale labels does not work when there is a lot of values. Showing every 10th (or another interval) label is a good idea. Alternatively, have a input field that let's the scientist choose their own "interval".

I should have mentioned the statistics. I have to rewrite some things for the statistics, and decided it was not a reason to hold off updating the website while a worked on that.

Thanks for the feedback!

**Comment by davitgigilashvili** [comment]
Yes, having a scientist a possibility to choose own interval is even better option. But, indeed, neither this, nor the statistics is urgent for now. All statistics can be easily calculated in the Excel sheet directly. So, it was a right decision to launch the Magnitude Estimation, as soon as it was ready. 

Thank you very much once again!

**Comment by disastersystem** [comment]
@davitgigilashvili 
I've added the matching experiment and called it "Match Estimation" (not sure why I didn't call it "Matching"). Hopefully it's not too late to be useful. Here is how it works:

The reference/original image of the image set will be used as the fixed image. The rest of the images in the set will be used as the variable image. So you’ll have to make an image set for every “page” in the experiment. I'm sure there are better ways to organize this, but it was what I could come up with at the moment. Let me know if you think of ways it can be improved.

The order of the variable image will be determined by the order of the images in the image set itself (under "Your stimuli groups"). I have added the possibility to rearrange the order of the images in the image sets by dragging them around. Image sets uploaded prior to this feature may not show in desired order, to fix this just move one image and an order will be set.

Limitations:
- You can't display the reference/original alongside like in other experiments.
- You can't have different sized image sets in the same experiment. Because the amount of slider steps is fixed, and the amount of variable images has to match the amount of slider steps.
- Moving the slider can be a bit "sluggish" if the user's internet is slow or images are large, since images are loaded on-demand rather than all at initial load. I think this should be changed to improve the user experience, and I plan to change this.

**Comment by davitgigilashvili** [comment]
Hi, thank you very much for implementing this!

Yes, it would be great if the slider could move more smoothly, but looks fine other than that. 
Maybe, just one more thing: is it possible to customize which image appears on the left and which one appears on the right? (right now the variable image with a slider is on the left and the fixed one is on the right).

**Comment by disastersystem** [comment]
The slider should move smoothly now.

There is no way to customize which image appears on the left or right, but thinking about it, having the fixed image on the left side might make more sense in general. So I changed the fixed image default position to be on the left side instead.

I've been considering some different features to make the experiment layout more flexible in general, changing the left/right position would fit under this.
I will close this issue soon, and make a new feature request issue regarding some layout changes.

**Comment by davitgigilashvili** [comment]
Great. Thank you very much!
